src/main/java/org/haffson/adventofcode/days/day02/Day02.java

```diff
@@ -0,0 +1,177 @@
+package org.haffson.adventofcode.days.day02;
+
+import org.haffson.adventofcode.ProblemStatusEnum;
+import org.haffson.adventofcode.days.Days;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Scanner;
+
+
+
+/**
+ * Implementation for <i>Day 1: Chronal Calibration</i>.
+ */
+@Component
+public class Day02 implements Days {
+
+    private final String path = "/Users/jenni/dedica/AdventOfCode/Jenni/Day02/src/input.txt";
+    private List<String> rawData = getRawDataAsList(path);
+
+    /** The puzzle status {@code HashMap} */
+    private final HashMap<String, ProblemStatusEnum> problemStatus;
+
+    @Autowired
+    Day02() {
+        this.problemStatus = new HashMap<>();
+        this.problemStatus.put("1", ProblemStatusEnum.SOLVED);
+        this.problemStatus.put("2", ProblemStatusEnum.SOLVED);
+    }
+
+    @Override
+    public int getDay() {
+        return 2;
+    }
+
+    @Override
+    public HashMap<String, ProblemStatusEnum> getProblemStatus() {
+        return problemStatus;
+    }
+
+    @Override
+    public String firstPart() {
+        try {
+            return "Part 1 answer: " + getNumberPwd1();
+        } catch (FileNotFoundException e) {
+            return "Error in method 1";
+        }
+    }
+
+    @Override
+    public String secondPart() {
+        try {
+            return "Part 2 answer: " + getNumberPwd2();
+        } catch (FileNotFoundException e) {
+            return "Error in method 2";
+        }
+    }
+
+
+    /** Method to read raw data from file into list
+     *
+     * @return raw data as list
+     */
+    public List<String> getRawDataAsList(String path) {
+        List<String> rawData = new ArrayList<>();
+
+        try {
+            Scanner s = new Scanner(new File(path)).useDelimiter("\n");
+            while (s.hasNext()) {
+                rawData.add(s.next());
+            }
+            s.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found!");
+        }
+        return rawData;
+    }
+
+    /**
+     * Method getData() to get min and max number, letter and password from raw data
+     * First, create class Data to avoid having multiple return values
+     */
+    public static class Data {
+        int min;
+        int max;
+        String letter;
+        String pwd;
+
+        public Data(int min, int max, String letter, String pwd) {
+            this.min = min;
+            this.max = max;
+            this.letter = letter;
+            this.pwd = pwd;
```

Comment:
Let us think about some more things here. What would happen with the business logic in your methods if the parameters were `null`?

Can `int` even be `null`? And `String` can't only be `null`, it can also be empty. 

One good place to validate this is the constructor.  In Java, I have seen it done quite frequent like this:
```suggestion
        public Data(int min, int max, @NonNull String letter, @NonNull String pwd) {
            this.min = min;
            this.max = max;
            this.letter = Objects.requireNonNull(letter);
            this.pwd = Objects.requireNonNull(pwd);
```
Then, you could also annotate your class parameters `letter` and `pwd` with `@NonNull`.

To check if the `String` is not empty, apache commons `StringUtil.isEmpty()` is used quite often.

Probably you would need to add apache commons to your dependency list in `build.gradle`. Check the first result of this google [search ](https://www.google.com/search?q=apache+commons+stringutils+maven&rlz=1C1GCEB_deDE882DE882&sxsrf=ALeKk02GVUb8UhfSNzAvqWd2MxxsHtO1AA%3A1627851902335&ei=fgwHYYXiE8v_7_UPk8qSMA&oq=apache+commons+stringutils+maven&gs_lcp=Cgdnd3Mtd2l6EAMyBQgAEMsBMgYIABAWEB4yBggAEBYQHjIGCAAQFhAeMgYIABAWEB4yBggAEBYQHjIGCAAQFhAeMgYIABAWEB4yBggAEBYQHjIGCAAQFhAeOgcIIxCwAxAnOgcIABBHELADOgcIABCwAxBDOggIABAWEAoQHkoFCDwSATFKBAhBGABQ5xhY2BxgkR5oAXACeACAAU-IAdIDkgEBN5gBAKABAcgBCsABAQ&sclient=gws-wiz&ved=0ahUKEwiFmYrz3JDyAhXL_7sIHROlBAYQ4dUDCA4&uact=5) that points towards the maven repository. This is the place, where most commonly, libraries are downloaded from. Pick a version and let it show you the gradle command you can directly copy into your `build.gradle`. After importing the dependency, you should be able to use `StringUtils` from your `Day02` class!